### PR TITLE
Change to ensure header files can be compiled with gcc -Wold-style-cast -Wall -pedantic -Wextra -Werror.

### PR DESCRIPTION
**Pull request by tepperly**
I compiled RAJA with  -DCMAKE_CXX_FLAGS="-Wall -pedantic -Werror -Wextra -Wold-style-cast" and resolved the issues found in header files.

### CODE REVIEW

**Comment by tepperly** [comment]
One alternative to -32768 would probably be reinterpret_cast<int32_t>(0xFFFF).  The issue is that 0xFFFF is technically an unsigned with value 65535, and it's an overview when casting it to a signed 32 bit.

File: include/RAJA/operators.hxx [hunk 1]
```diff
@@ -245,11 +245,11 @@ template <typename T>
 RAJA_HOST_DEVICE constexpr T min()
 {
   return 0;
-};
+}
 template <>
 RAJA_HOST_DEVICE constexpr int16_t min<int16_t>()
 {
-  return 0xFFFF;
+  return -32768;
```
Comment (davidbeckingsale) [inline include/RAJA/operators.hxx]: Later in the file, macros from float.h are used (like FLT_MAX). There are analogous macros for integral types defined in limits.h that would probably be more appropriate here.

Comment (rhornung67) [inline include/RAJA/operators.hxx]: I think it's best to use standard definitions for such constants. We just need to make sure it doesn't break portability; e.g., probably stay away from POSIX, does it work with CUDA, etc.

Comment (tepperly) [inline include/RAJA/operators.hxx]: This demonstrates two ways to do it:

    #include <iostream>
    #include <cstdint>
    #include <limits>
    
    int main(int argc, char **argv) {
      std::cout << "Max int16 via macro: " << INT16_MIN
                << "\nMax int16 via numeric_limits: "
                << std::numeric_limits<int16_t>::min()
                << "\n";
    }

Comment (tepperly) [inline include/RAJA/operators.hxx]: Presumably my -32768 is no worse than 0xFFFF.

Comment (tepperly) [inline include/RAJA/operators.hxx]: If I make more commits to my branch does the show up in this pull request?  Do you want me to update my pull request? I won't be offended if someone else gets the fix in place. I was just trying to be helpful.

Comment (davidbeckingsale) [inline include/RAJA/operators.hxx]: Yup, you can push more commits to the branch and they will show up. We're really grateful for the PR!

Personally, I like the `numeric_limits` approach.

Comment (rhornung67) [inline include/RAJA/operators.hxx]: @tepperly thanks much for helping out. Having RAJA development evolve as a community effort is something we want. I agree with @davidbeckingsale comment about the numeric_limits approach.

**Comment by davidbeckingsale** [comment]
Thanks very much @tepperly 😄

**Pull request merged**